src/test/java/org/glite/authz/pep/pip/provider/PolicyNamesPIPTest.java

```diff
@@ -0,0 +1,197 @@
+ * See http://www.eu-egee.org/partners/ for details on the copyright holders.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Authors:
+ * 2016-
+ * Mischa Salle <msalle@nikhef.nl>
+ * NIKHEF Amsterdam, the Netherlands
+ * <grid-mw-security@nikhef.nl>
+ */
+
+package org.glite.authz.pep.pip.provider;
+
+import org.glite.authz.common.model.Request;
+import org.glite.authz.common.model.Subject;
+import org.glite.authz.common.model.Attribute;
+import org.glite.authz.pep.pip.PIPProcessingException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.After;
+
+/**
+ * JUnit framework tests for {@link PolicyNamesPIP}
+ * @author Mischa Sall&eacute;
+ */
+public class PolicyNamesPIPTest {
+    PolicyNamesPIP pip;
+
+    private final Logger log = LoggerFactory.getLogger(PolicyNamesPIP.class);
+
+    private final static String GoodDN = "/C=NL/O=Example/OU=PDP/CN=Test CA";
+    
+    private final static String BadDN = "/C=John Doe";
+
+    /**
+     * Setup a new {@link PolicyNamesPIP_PIP} PIP and set the accepted attributes
+     * to all of them
+     */
+    @Before
+    public void initialize() throws Exception {
+	log.debug("Creating PIP");
+	pip = new PolicyNamesPIP("PolicyNamesPIP", getClass().getResource("/certificates").getFile());
+        pip.start();
+    }
+
+    /** Stop the {@link PolicyNamesPIP} PIP */
+    @After
+    public void finalize() throws Exception {
+	log.debug("Stopping PIP");
+	if (pip!=null)
+	    pip.stop();
+	else
+	    log.warn("PIP already stopped it seems...");
+    }
+
+    /**
+     * Helper method for handling tests that throw a PIPProcessingException
+     * @param request input request
+     * @param element String which should match the stacktrace message
+     * @return boolean: true when PIP has applied, false otherwise
+     */
+    protected boolean throwsPIPProcessingException(Request request, String element) {
+	    boolean applied= false;
+	    try {
+		log.debug("Request="+request);
+		applied= pip.populateRequest(request);
+		fail("must throw PIPProcessingException");
+	    } catch (PIPProcessingException e) {
+		// expected
+		String message= e.getMessage();
+		log.debug("EXPECTED: " + message);
+		log.debug("GOT ("+message.indexOf(element)+"): " + element);
+		assertTrue("PIPProcessingException message does not contain: " + element, message.indexOf(element) >= 0);
```

Comment:
agree we should have consistency, but my objective here is to ensure that new code that enters to the codebase satisfies certain requirements. So please try the migration to hamcrest, and if you have questions/doubts I will be happy to help